test/passing/tests/labeled_tuples_cmts_attrs_move.ml.ref

```diff
@@ -0,0 +1,26 @@
+(* Tests making sure comments and attributes are handled reasonably by labeled tuple
+   printing.  This test has where we allow ourselves to slightly move a comment or add a
+   pun or parens that aren't strictly necessary.  These examples are all analagous to
+   similar cases with other forms (e.g., labeled arguments).  See
+   [labeled_tuples_cmts_attrs.ml] for examples that stay exactly the same after
+   formatting. *)
+
+(* Attrs around expressions *)
+let y = ~z, (z [@attr])
+let y = ~z, ~z [@@attr]
+let y = ~z:((42 [@attr]) : int), 42
+
+(* Comments around expressions *)
+let _ = (* baz *) ~z, ~(y : int)
+let _ = ~z, ~(* baz *) (y : int)
+let _ = ~z, ~(* baz *) (y : int)
+
+(* Attrs around types *)
+type t = (z:int * y:bool[@attr])
+
+(* Comments around patterns *)
+let (* baz *) ~z, ~y = ()
+let ~z, (* baz *) ~y = ()
+let ~z:42, ~((* baz *) y : int) = ()
+let ~z:42, ~((* baz *) y : int) = ()
+let ~z:42, ~((* baz *) y : int) = ()
```

Comment:
Yes - this whole thing is very messy.  I believe the reason for the difference just has to do with subtle differences in the parser.  There are a couple ways it could be improved, but considering the rarity of these forms, it does not seem worth the effort.  I've integrated your suggestion below to at least make it consistent - thanks!